src/Flysystem/GeneratedAssets.php

```diff
@@ -102,21 +102,20 @@ protected function checkOrCreate($filename, $callback = null)
     public function setContent($filename, $content)
     {
         // Store content
-        $result = $this
-                ->getFilesystem()
-                ->put($filename, $content);
-
-        if (!$result) {
-            throw new Exception("Error regenerating file \"{$filename}\"");
+        try {
+            $this
+            ->getFilesystem()
+            ->write($filename, $content);
+        } catch (UnableToWriteFile $exception) {
+            throw new UnableToWriteFile("Error regenerating file \"{$filename}\"");
```

Comment:
```suggestion
        $result = $this
                ->getFilesystem()
                ->put($filename, $content);

        if (!$result) {
            throw new UnableToWriteFile("Error regenerating file \"{$filename}\"");
```
No need to create a new `UnableToWriteFile` after catching

Reply:
DONE